fix: update section headings to align with v1 documentation (#1343)
auth0/docs-v2: main/docs/api/authentication/signup/create-a-new-user.mdx

  The user metadata to be associated with the user. If set, the field must be an object containing no more than ten properties. Property names can have a maximum of 100 characters, and property values must be strings of no more than 500 characters.
</ParamField>

## Response Schema

<ResponseSchema>
<ResponseField name="_id" type="string">
  The user's unique identifier.
</ResponseField>

<ResponseField name="email_verified" type="boolean">

>   Whether the user's email address has been verified.

Reviewer: ```suggestion
  Indicates whether the user's email address is verified. 
```
